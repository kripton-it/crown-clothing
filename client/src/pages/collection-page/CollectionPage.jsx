import React from "react";
import { connect } from "react-redux";

import { selectCollection } from "../../redux/shop/shop-selectors";

import CollectionItemContainer from "../../components/collection-item/CollectionItemContainer";

import {
  CollectionPageContainer,
  TitleContainer,
  ItemsContainer
} from "./CollectionPageStyles";

const CollectionPage = ({ collection }) => {
  const { title, items } = collection;

  return (
    <CollectionPageContainer>
      <TitleContainer>{title}</TitleContainer>
      <ItemsContainer>
        {items.map(item => (
          <CollectionItemContainer
            className="collection-item"
            key={item.id}
            item={item}
          />
        ))}
      </ItemsContainer>
    </CollectionPageContainer>
  );
};

const mapStateToProps = (state, ownProps) => ({
  collection: selectCollection(ownProps.match.params.collectionId)(state)
});

export default connect(mapStateToProps)(CollectionPage);
