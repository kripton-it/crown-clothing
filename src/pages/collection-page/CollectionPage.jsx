import React from 'react';

import CollectionItem from '../../components/collection-item/CollectionItem'

// import './CollectionPage.styles.scss'

const CollectionPage = ({match}) => {
  return (
    <div className="collection-page">
      <h2>Category page</h2>
    </div>
  );
};

export default CollectionPage;