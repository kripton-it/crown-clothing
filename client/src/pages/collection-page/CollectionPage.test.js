import React from "react";
import { shallow } from "enzyme";

import { CollectionPage } from "./CollectionPage";

it("expect to render CollectionPage page", () => {
  const mockProps = {
    collection: {
      title: "Hello",
      items: [
        {
          id: 1,
          imageUrl: "abc.com",
          name: "Hello",
          price: 100,
          quantity: 2
        }
      ]
    }
  };
  expect(shallow(<CollectionPage {...mockProps} />)).toMatchSnapshot();
});
