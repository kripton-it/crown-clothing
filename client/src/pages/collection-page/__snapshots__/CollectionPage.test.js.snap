// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expect to render CollectionPage page 1`] = `ShallowWrapper {}`;
